### ISSUE TITLE
responseFile ignores given status, always sends 200 (or 404)

### ISSUE THREAD

**Issue by exergasia** [yesodweb/wai#527]
```
$ cat Test.hs
import Network.HTTP.Types.Status
import Network.Wai
import Network.Wai.Handler.Warp

main = run 9999 $ \ _ k -> k $ responseFile status400 [] "Test.hs" Nothing
$ runhaskell Test.hs &
[1] 3439
$ curl -v 'http://localhost:9999/'
* Adding handle: conn: 0x896b288
* Adding handle: send: 0
* Adding handle: recv: 0
* Curl_addHandleToPipeline: length: 1
* - Conn 0 (0x896b288) send_pipe: 1, recv_pipe: 0
* About to connect() to localhost port 9999 (#0)
*   Trying 127.0.0.1...
* Connected to localhost (127.0.0.1) port 9999 (#0)
> GET / HTTP/1.1
> User-Agent: curl/7.32.0
> Host: localhost:9999
> Accept: */*
> 
< HTTP/1.1 200 OK
< Last-Modified: Fri, 01 Apr 2016 12:30:58 GMT
< Content-Length: 160
< Accept-Ranges: bytes
< Date: Fri, 01 Apr 2016 12:31:35 GMT
* Server Warp/3.2.4 is not blacklisted
< Server: Warp/3.2.4
< 
import Network.HTTP.Types.Status
import Network.Wai
import Network.Wai.Handler.Warp

main = run 9999 $ \ _ k -> k $ responseFile status400 [] "Test.hs" Nothing
```

Not shown: it sends 404 if responseFile is passed a bogus filename. That should probably be 500, since it's an application whoopsie.

**Comment by exergasia** [comment]
Oh, I see that this is actually according to the documentation: Network.Wai.Handler.Warp.Response.sendResponse explicitly says that it will ignore the given status if the FilePart is Nothing. My contention is that that's really, really surprising and under-documented, and is a blatant foot-gun for anyone trying to serve error pages off disk. At the very least, Network.Wai.responseFile needs a great honking big warning, or else people will assume that, since it has a Status parameter, it will be respected (see also: this 'bug' report).

**Comment by kazu-yamamoto** [comment]
I agree that it is hard to understand this API. Unfortunately, it's too late to change the API since there are so many programs which use `responseFile`. So, I would take a big warning. Would you send us a pull request?

**Comment by seagreen** [comment]
The documentation for this in in [warp](https://hackage.haskell.org/package/warp-3.2.18.2/docs/Network-Wai-Handler-Warp-Internal.html#v:sendResponse), there's none at all in `wai` where [responseFile](https://hackage.haskell.org/package/wai-3.2.1.2/docs/Network-Wai.html#v:responseFile) is defined. That's definitely not ideal.

What's the connection between providing a `FilePart` and the response status being chosen automatically? I'd like to help with the docs, but I don't understand the behavior well enough to do so yet.
